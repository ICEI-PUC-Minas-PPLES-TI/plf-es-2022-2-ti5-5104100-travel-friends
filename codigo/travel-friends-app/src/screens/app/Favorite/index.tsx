import React from "react";
import { View, Text, Image, TouchableOpacity } from "react-native";
import theme from "../../../utils/theme";
import { style } from "./styles";

export function Favorite() {
  return (
    <View style={style.container}>
      <View style={style.logo}>
        <Text
          style={[
            style.title,
            { color: theme.colors.title, fontWeight: "600" },
          ]}
        >
          Favorite
        </Text>
      </View>
    </View>
  );
}
