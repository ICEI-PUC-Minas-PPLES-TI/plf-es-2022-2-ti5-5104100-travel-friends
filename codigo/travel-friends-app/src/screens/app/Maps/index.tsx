import React, { useState } from "react";
import { View } from "react-native";
import MapView, { Marker } from "react-native-maps";

import { style } from "./styles";

export function Maps() {
  const [region, setRegion] = useState(null);

  const latitude = -19.9337364;
  const longitude = -43.9362059;

  return (
    <View style={style.container}>
      <MapView
        initialRegion={{
          latitude: latitude,
          longitude: longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }}
        style={style.map}
        loadingEnabled
        showsUserLocation
      >
        <Marker coordinate={{ latitude: latitude, longitude: longitude }} />
      </MapView>
    </View>
  );
}
//TODO: colocar os pins do mapas dos roteiros
