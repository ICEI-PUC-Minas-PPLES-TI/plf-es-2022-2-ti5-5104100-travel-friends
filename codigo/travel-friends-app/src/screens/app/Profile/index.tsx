import { FontAwesome } from "@expo/vector-icons";
import React, { useEffect } from "react";
import { View, TouchableOpacity, StatusBar } from "react-native";
import TextApp from "../../../components/Text";
import { formatNickName } from "../../../utils/formatNickName";
import theme from "../../../utils/theme";
import { useAuth } from "../../../contexts/useAuth";
import { userHook } from "../../../contexts/userData";

import { styles } from "./styles";
import { useTabAPPNavigation } from "../../../hooks/navigation";
import { useNotification } from "../../../contexts/useNotification";
import { useFocusEffect } from "@react-navigation/native";

export function Profile() {
  const { logout } = useAuth();
  const { userData } = userHook();
  const { notificatioList } = useNotification();
  const navigation = useTabAPPNavigation();

  const not = notificatioList.length;

  useFocusEffect(() => {
    console.log("aaa");
  });

  return (
    <View style={styles.container}>
      <View style={styles.headerProfile}>
        <View style={styles.nickName}>
          <TextApp
            text={formatNickName(userData.name)}
            size={24}
            color={theme.colors.shape}
          />
        </View>
        <View style={styles.underlineProfile}>
          <TextApp text={userData.name} size={24} />
        </View>
      </View>
      <View style={styles.bodyProfile}>
        <View style={styles.menuProfile}>
          <TouchableOpacity
            onPress={() => navigation.navigate("FavoriteScreens")}
          >
            <View style={styles.underlineButton}>
              <FontAwesome
                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                //@ts-ignore
                name={"heart"}
                size={24}
                color={theme.colors.title}
                style={{ marginRight: 20 }}
              />
              <TextApp text="Favoritos" size={theme.fonts.text} />
            </View>
          </TouchableOpacity>

          <TouchableOpacity>
            <View style={styles.underlineButton}>
              <FontAwesome
                name={"bell"}
                size={24}
                color={theme.colors.title}
                style={{ marginRight: 20 }}
              />
              <TextApp text={`Notificação - ${not}`} size={theme.fonts.text} />
            </View>
          </TouchableOpacity>

          <TouchableOpacity>
            <View style={styles.underlineButton}>
              <FontAwesome
                name={"user"}
                size={24}
                color={theme.colors.title}
                style={{ marginRight: 20 }}
              />
              <TextApp text="Editar conta" size={theme.fonts.text} />
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => logout()}>
            <View style={styles.underlineButton}>
              <FontAwesome
                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                //@ts-ignore
                name={"arrow-circle-o-right"}
                size={24}
                color={theme.colors.title}
                style={{ marginRight: 20 }}
              />
              <TextApp
                text="sair"
                size={theme.fonts.text}
                color={theme.colors.title}
              />
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}
