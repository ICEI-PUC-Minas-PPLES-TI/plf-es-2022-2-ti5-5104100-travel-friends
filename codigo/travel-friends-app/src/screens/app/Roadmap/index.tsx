import React, { useState } from "react";
import { RouteProp, useRoute } from "@react-navigation/native";
import { View } from "react-native";
import MapView, { Marker } from "react-native-maps";
import Button from "../../../components/Button";
import TextApp from "../../../components/Text";
import theme from "../../../utils/theme";

import { style } from "./styles";
import { HomeRoutesParams } from "../../../routes/routes";
import HeaderGoback from "../../../components/HeaderGoback";

type RoadmapScreenProps = RouteProp<HomeRoutesParams, "Roadmap">;

export default function Roadmap() {
  const { params } = useRoute<RoadmapScreenProps>();
  const latitude = -19.9337364;
  const longitude = -43.9362059;

  return (
    <View style={style.container}>
      <HeaderGoback color={theme.colors.text} title={params.title} />
      <MapView
        initialRegion={{
          latitude: latitude,
          longitude: longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }}
        style={style.map}
        loadingEnabled
        showsUserLocation
      >
        <Marker coordinate={{ latitude: latitude, longitude: longitude }} />
      </MapView>
      <View style={style.content}>
        <View style={style.header}>
          <View>
            <TextApp
              size={theme.fonts.title}
              text={params?.title}
              isBold
              fontWeight="700"
              color={theme.colors.title}
            />
            <TextApp
              size={theme.fonts.subText}
              text={` ${params.start} - ${params.end}`}
              color={theme.colors.title}
              fontWeight="400"
            />
          </View>
          <View style={style.headerPrice}>
            <TextApp
              size={theme.fonts.subTitle}
              text={params.isFree ? "Gratuito" : ` R$ ${params.price}`}
              color={theme.colors.primary_30}
              fontWeight="400"
              isBold
            />
          </View>
        </View>
        <View style={style.body}>
          <TextApp
            size={theme.fonts.subTitle}
            text={"Descrição"}
            color={theme.colors.title}
            isBold
          />
          <TextApp
            size={theme.fonts.subText}
            text={`${params?.description}`}
            color={theme.colors.text}
          />
          <View style={{ paddingTop: 16 }}>
            <TextApp
              size={theme.fonts.subTitle}
              text={"Participantes"}
              color={theme.colors.title}
              isBold
            />
            <TextApp
              size={theme.fonts.subText}
              text={` ${params.participants.length} de ${params.person} `}
              color={theme.colors.text}
            />
          </View>
        </View>
        <View style={style.footer}>
          <Button
            onPress={() => {
              console.log(params);
            }}
            title="Inscreva-se"
          />
        </View>
      </View>
    </View>
  );
}
