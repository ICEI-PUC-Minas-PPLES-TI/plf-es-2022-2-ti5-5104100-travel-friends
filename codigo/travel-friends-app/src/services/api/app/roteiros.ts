import api from "../api";

const url = "https://travel-friends-backend.herokuapp.com/";

export async function getAllRoadmap() {
  try {
    const response = await api.get(url);

    const data = {
      data: response.data,
      status: response.status,
    };

    console.log(data);
    return data;
  } catch (error) {
    console.log("---> Roteiro error: ");
  }
}
