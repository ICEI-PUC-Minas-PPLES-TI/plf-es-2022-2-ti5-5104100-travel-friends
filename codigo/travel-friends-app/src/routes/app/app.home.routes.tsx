import React from "react";

import { createStackNavigator } from "@react-navigation/stack";
import { HomeRoutesParams } from "../routes";

import Home from "../../screens/app/Home";
import Roadmap from "../../screens/app/Roadmap";
const { Navigator, Screen } = createStackNavigator<HomeRoutesParams>();

export function HomeRoutes() {
  return (
    <Navigator
      initialRouteName={"Home"}
      screenOptions={{
        headerShown: false,
      }}
    >
      <Screen name="Home" component={Home} />
      <Screen name="Roadmap" component={Roadmap} />
    </Navigator>
  );
}
