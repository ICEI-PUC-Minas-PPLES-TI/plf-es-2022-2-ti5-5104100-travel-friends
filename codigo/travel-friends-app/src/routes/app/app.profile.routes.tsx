import React from "react";

import { createStackNavigator } from "@react-navigation/stack";
import { ProfileRoutesParams } from "../routes";

import { Profile } from "../../screens/app/Profile";

const { Navigator, Screen } = createStackNavigator<ProfileRoutesParams>();

export function ProfileRoutes() {
  return (
    <Navigator
      initialRouteName={"Profile"}
      screenOptions={{
        headerShown: false,
      }}
    >
      <Screen name="Profile" component={Profile} />
    </Navigator>
  );
}
