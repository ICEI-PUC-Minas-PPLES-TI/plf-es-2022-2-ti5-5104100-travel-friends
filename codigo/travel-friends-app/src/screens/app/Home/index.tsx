import React, { useEffect, useState } from "react";
import { View, FlatList, Dimensions, StatusBar } from "react-native";
import { userHook } from "../../../contexts/userData";
import { RoadMap } from "../../../@types/models.interface";
import { getAllRoadmap } from "../../../services/api/app/roteiros";

import Header from "../../../components/Header";
import CardInfo from "../../../components/CardInfo";

import { style } from "./styles";
import theme from "../../../utils/theme";
import TextApp from "../../../components/Text";

export default function Home() {
  const user = userHook();
  const { height, width } = Dimensions.get("window");

  const [roadmapList, setRoadmapList] = useState<RoadMap[]>([]);

  async function loadRoadmap() {
    const resp = await getAllRoadmap();
    setRoadmapList(resp?.data);
  }

  function renderItem(item: RoadMap) {
    return <CardInfo item={item} />;
  }

  useEffect(() => {
    loadRoadmap();
  }, []);
  console.log(roadmapList);
  const { userData } = user;
  return (
    <View style={style.container}>
      <StatusBar
        barStyle="light-content"
        backgroundColor={theme.colors.primary}
      />
      <Header />

      <View style={{}}>
        <FlatList
          horizontal
          showsHorizontalScrollIndicator={false}
          snapToAlignment={"start"}
          scrollEventThrottle={16}
          decelerationRate={"fast"}
          keyExtractor={(item, i) => `${item}${i}`}
          snapToOffsets={[...Array(roadmapList.length)].map(
            (item, index) => index * (width * 0.8 - 40) + (index - 1) * 40
          )}
          data={roadmapList}
          renderItem={({ item }) => renderItem(item)}
          style={{ marginTop: -40 }}
        />
      </View>

      <View style={{ paddingHorizontal: 20 }}>
        <TextApp
          size={theme.fonts.subTitle}
          text={"Em alta"}
          isBold
          color={theme.colors.title}
        />
      </View>
    </View>
  );
}
