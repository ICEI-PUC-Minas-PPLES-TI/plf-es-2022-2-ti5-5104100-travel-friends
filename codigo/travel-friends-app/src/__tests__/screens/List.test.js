import React from "react";
import renderer from "react-test-renderer";

import { List } from "../../screens/app/List";

describe("<List />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<List />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<List />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
