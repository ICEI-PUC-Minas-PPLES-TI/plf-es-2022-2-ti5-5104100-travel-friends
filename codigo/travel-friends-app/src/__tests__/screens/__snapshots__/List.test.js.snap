// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fafcfd",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 30,
        "width": 750,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fafcfd",
            "fontSize": 24,
            "fontWeight": "600",
          },
          Object {
            "color": "#101d42",
            "fontWeight": "600",
          },
        ]
      }
    >
      Agenda
    </Text>
  </View>
</View>
`;
